dali/dfu/dfurun.cpp

```diff
@@ -1460,9 +1473,16 @@ class CDFUengine: public CInterface, implements IDFUengine
                                 Audit("COPYDIFF",userdesc,srcName.get(),dstName.get());
                             }
                         }
-                        else if (foreigncopy||auxfdesc) {
-                            fsys.import(auxfdesc.get()?auxfdesc.get():foreignfdesc.get(), dstFile, recovery, recoveryconn, filter, opttree, &feedback, &abortnotify, dfuwuid);
-                            if (!abortnotify.abortRequested()) {
+                        else if (foreigncopy||auxfdesc)
+                        {
+                            IFileDescriptor * srcDesc = (auxfdesc.get() ? auxfdesc.get() : foreignfdesc.get());
+                            if (options->getPreserveCompression() && srcDesc && srcDesc->isCompressed())
+                                dstFile->queryAttributes().setPropBool("@blockCompressed",true);
```

Comment:
@AttilaVamos - This will get the right result, but it should be setup before it gets here.
i.e. dstFile and the IFileDescriptor is was based, as with other cases should be preset to correct compression before it gets here.

I'm assuming this was added because it doesn't get set on lines 1326-27..?
If so, why didn't it those lines?

Reply:
It hit those lines and crashed there because srcFile is a distributed file (for copy) and it wasn't initialized for remote copy (that's why I add pointer checks '&& srcFile' there),

Reply:
@AttilaVamos - the check should still be before and set the target fdesc (on line 1327) as with other cases.
So it should be there that it checks foreignfdesc or auxfdesc if no srcFile is available.

Reply:
this is what I think code should look like (untested):

replace lines 1326-1327
with:
```
    else if (options->getPreserveCompression())
    {
        bool dstCompressed = false;
        if (srcFile)
            dstCompressed = srcFile->isCompressed();
        else
        {
            IFileDescriptor * srcDesc = (auxfdesc.get() ? auxfdesc.get() : foreignfdesc.get());
            dstCompressed = srcDesc && srcDesc->isCompressed();
        }
        if (dstCompressed)
            fdesc->queryProperties().setPropBool("@blockCompressed", dstCompressed);
    }
```